### PR TITLE
content: make the content cache a broker module

### PR DESCRIPTION
**Pull request by garlick**
Problem: the content service embedded in the broker is resource-intensive under load, potentially impacting broker performance.  Furthermore, it is a complex, optional service rather than an integral broker component.
    
Move the content service to a broker module where it is less tightly coupled to the broker and has a private reactor thread.

I didn't see any obvious impact on performance when running the throughput test.

**This branch**
```
$ src/test/throughput.py -n 10000
number of jobs: 10000
submit time:    2.964 s (3373.6 job/s)
script runtime: 45.773s
job runtime:    45.616s
throughput:     219.2 job/s (script: 218.5 job/s)

$ src/test/throughput.py -n 10000 -x
number of jobs: 10000
submit time:    3.216 s (3109.4 job/s)
script runtime: 449.663s
job runtime:    103.558s
throughput:     96.6 job/s (script:  22.2 job/s)
```

**Master (v0.54.0-32-g9d9216681)**
```
$ src/test/throughput.py -n 10000
number of jobs: 10000
submit time:    3.008 s (3324.5 job/s)
script runtime: 46.227s
job runtime:    46.054s
throughput:     217.1 job/s (script: 216.3 job/s)

$ src/test/throughput.py -n 10000 -x
number of jobs: 10000
submit time:    2.850 s (3509.2 job/s)
script runtime: 460.330s
job runtime:    93.438s
throughput:     107.0 job/s (script:  21.7 job/s)
```

### CODE REVIEW

**Comment by garlick** [comment]
The accounting test won't pass until flux-framework/flux-accounting#383 is merged.

File: t/t0011-content-cache.t [hunk 1]
```diff
@@ -63,7 +66,7 @@ test_expect_success 'store test blobs on rank 0' '
 	flux content store <1m.0.store >1m.0.hash
 '
 
-test_expect_success LONGTEST "cannot store blob that exceeds max size of $MAXBLOB" '
+test_expect_success "cannot store blob that exceeds max size of $MAXBLOB" '
```
Comment (chu11) [inline t/t0011-content-cache.t:69]: just want to make sure removal of LONGTEST intentional?

File: t/t1003-kvs-stress.t [hunk 2]
```diff
@@ -97,18 +97,10 @@ test_expect_success 'kvs: 8 threads/rank each doing 100 put,fence in a loop, mix
...
-'
-
 test_expect_success 'kvs: store 10,000 keys in one dir' '
 	${FLUX_BUILD_DIR}/t/kvs/torture --prefix $DIR.bigdir --count 10000
 '
 
-test_expect_success LONGTEST 'kvs: store 100,000 keys in one dir' '
-	${FLUX_BUILD_DIR}/t/kvs/torture --prefix $DIR.bigdir2 --count 100000
-'
-
```
Comment (chu11) [inline t/t1003-kvs-stress.t:111]: perhaps update commit message to also say that some tests were removed due to blob size limit.  Was just initially confused why some tests were outright removed.

**Review by chu11 — APPROVED** [review]
overall LGTM, just small comments below.  I'm surprised how little changes were necessary with this move.  I thought it'd be a ton more.

**Comment by garlick** [comment]
Now that flux-framework/flux-accounting#383 is merged I'll set MWP here.  Thanks!

**Pull request merged**